File: HFSGuardaDiretorio_Java/hfsguardadir/src/main/java/hfsguardadir/objetos/Tipo.java
package hfsguardadir.objetos;

import hfsguardadir.comum.Rotinas;

public class Tipo {

    private char codigo;

    private String nome;

    public Tipo() {
        this.limparDados();
    }

    public Tipo(char codigo, String nome) {
        this.codigo = codigo;
        this.nome = nome;
    }

    public char getCodigo() {
        return codigo;
    }

    public void setCodigo(char codigo) {
        this.codigo = codigo;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public final void limparDados() {
        this.codigo = ' ';
        this.nome = "";
    }

    public boolean equals(Tipo obj) {
        return obj.getCodigo() == this.getCodigo()
                && obj.getNome().equals(this.getNome());
    }

    @Override
    public String toString() {
        return Rotinas.getRecurso("Tipo.codigo") + " " + this.codigo + ", "
                + Rotinas.getRecurso("Tipo.nome") + " " + this.nome;
    }

}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_FoxTookit/hfsguardadir/objetos/Tipo.cpp
#include "Tipo.h"

namespace hfsguardadir {

namespace objetos {

Tipo::Tipo()
{
    this->limparDados();
}

Tipo::Tipo(const char &codigo, const std::string &nome) {
    this->codigo = codigo;
    this->nome = nome;
}

char &Tipo::getCodigo() {
    return codigo;
}

void Tipo::setCodigo(const char &codigo) {
    this->codigo = codigo;
}

std::string &Tipo::getNome() {
    return nome;
}

void Tipo::setNome(const std::string &nome) {
    this->nome = nome;
}

void Tipo::limparDados() {
    this->codigo = ' ';
    this->nome = "";
}

bool Tipo::equals(Tipo obj) {
    return obj.getCodigo() == this->getCodigo()
        && obj.getNome() == this->getNome();
}

std::string &Tipo::toString() {
    return this->nome;
}

}
}
